pkg/controller/servicebindingrequest/annotations/resource.go

```diff
@@ -0,0 +1,204 @@
+package annotations
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/nested"
+)
+
+// ResourceHandler handles annotations related to external resources.
+type ResourceHandler struct {
+	// bindingInfo contains the binding details related to the annotation handler.
+	bindingInfo *BindingInfo
+	// client is the client used to retrieve a related secret.
+	client dynamic.Interface
+	// relatedGroupVersionResource is the related resource GVR, used to retrieve the related resource
+	// using the client.
+	relatedGroupVersionResource schema.GroupVersionResource
+	// relatedResourceName is the name of the related resource that is referenced by the handler
+	// annotation key/value pair.
+	relatedResourceName string
+	// resource is the unstructured object to extract data using inputPath.
+	resource unstructured.Unstructured
+	// valueDecoder is a function used to decode values from the resource being handled; for example,
+	// to decode Base64 keys the decodeBase64String can be used.
+	valueDecoder func(interface{}) (string, error)
```

Comment:
IMHO `stringValue` indicates better what is the intention, i.e. we need to get a string representation for the given interface, or an error.

Reply:
Solved in ea30c9d16fdf53b3830bb35a9fe0cdfc2eca2609.